advanced_topics.qmd

```diff
@@ -266,15 +277,359 @@ pums_person |>
   collect()
 ```
 
+### Using UDFs for regression model predictions
+
+Another example where UDFs can be useful is where you've created a model based on a subset of your data and you want to use it to make predictions.
+Let's say we've created a linear model, using education and employment status to predict hours worked per week for respondents in 2018.
+
+```{r}
+#| label: model-commute-time
+model_data <- open_dataset("data/person/") |>
+    filter(year == 2018) |>
+    filter(AGEP >= 16, AGEP < 65, !is.na(COW)) |>
+    transmute(
+      self_employed = str_detect(as.character(COW), "Self-employed"),
+      edu = case_when(
+        grepl("bachelor|associate|professional|doctorate|master", SCHL, TRUE) ~
+          "College or higher",
+        grepl("college", SCHL, TRUE) ~
+          "Some college",
+        grepl("High school graduate|high school diploma|GED", SCHL, TRUE) ~
+          "High school",
+        TRUE ~ "Less than High school"
+      ),
+      hours_worked = WKHP,
+      PWGTP = PWGTP
+    ) |>
+    collect()
+
+hours_worked_model <- survey::svyglm(
```

Comment:
We should explain using the `survey` package I think--I assume that's for the weighted regression?

Reply:
Ack yes definitely. I had on my mental todo list to do that but didn't get to it. It is indeed to do the weighted regression

Reply:
Done (not yet pushed)